### DISCUSSION TITLE
Add name/label to widget (and main screen)

### DISCUSSION THREAD

**Discussion by savek-cc** [Navid200/xDrip#159] [Ideas]
When using one phone to monitor the BG of multiple persons (e.g. 2 children), it would be nice to be able assign each a label (e.g. Name) - and also have this label visible on the xDrip main screen and startscreen widget.
(Using xDrip-Variants in this scenario).
Have been looking at the widget sources and couldn't find any option for a label.

**Comment by Navid200** [comment]
Go to Settings -> xDrip+ Display Settings -> xDrip+ Color Settings
Scroll all the way down to "Flair colors".
Enable it.
Change the upper title bar flair and lower button bar flair colors.

Do the same for the other variant.  But, choose different colors.

Now, you will see the colors you chose at the top and bottom of the main screen.

On the same page, you can also find "Widget chart background color".
Again, you should be able to give distinct colors to your different variants.

So, no we don't have labels.  But, using colors, you should be able to accomplish what you desire.

**Reply by savek-cc** [reply]
In theory yes - but you already have the widget position, so "top" is "Sandy" and "bottom" is Julie.
It doesn't really help much to also have another mapping "red" is "Sandy" and "yellow" is Julie - it's just another abstraction.
I'm currently looking at extending the status line - so it would be enabled by setting the status line for the widget with a freetext input for a label.
Or we just need to call out T1D kids by color names to get used to it... Hey blue! Don't hit yellow with red's shovel!!

**Reply by savek-cc** [reply]
![Screenshot_20211218-001821_Pixel_Launcher~2](https://user-images.githubusercontent.com/13501427/146618603-b0eb8010-0f78-4890-8fc4-ab6888591ce1.png)
Looks like this without further modification

**Reply by Navid200** [reply]
There is only so far we can go with this.
Let's say you add an option to the status line so that you can place a label on it.  So, you can see a name on the main screen and for each variant, you use the corresponding name.

But, then, you go to the settings to set up Nightscout and you forget which kid's xDrip you were working on.  The status line is only shown on the main screen.  As soon as you go to settings, and there are hundreds of settings pages in xDrip, there will be no way to know which one you are in.
So, there will always be a situation where you will need to back out and confirm you are where you think you are.


So, my question for you is, are you sure you will be happy if you get a label on the main screen?  Because I'm sure as soon as we have that, someone is going to ask for the label to be visible on every page, not just the main screen.

Please note that xDrip already has too many settings.  Every new feature is an addition to what needs to be maintained.  Just because we have been maintaining this app, with all the different settings, doesn't mean that it is easy.

Sometimes, we have to choose our battles.
There are so many bugs that we need to fix.  So, we are really trying to lower the priority of the feature requests unless there is an obvious necessity.

**Reply by savek-cc** [reply]
I understand your concern... back when I was working on some parts of xDrip+ with either Adrian or Jamorham, this "feature-explosion" was one of tha main reasons why a lot of ppl chose xDrip+ over xDrip. But I do understand that the userbase has grown since - so I'll just add it to my own branch (together with the miband-pr for that matter) and leave it at that.
Getting quite commercial here - I guess the good old "hacking" days are gone as too many ppl now depend on this and are overwhelmed by all the possible configurations and settings.
P.S.: Maybe time to fork again and do a "yDrip" for the not faint at heart? ;)
P.P.S.: OMG - those hacking days are already 5 years in the past....

**Comment by Navid200** [comment]
We have discussions now opened in the main repository:
https://github.com/NightscoutFoundation/xDrip/discussions

I am looking for a way to transfer the open discussions from here to there. But, I'm not sure I'll succeed. In that case, it may be easier to open new discussions there.

**Comment by Navid200** [comment]
@savek-cc If you have already spent time and developed this feature, you can open a PR.
We suggest people to open a discussion before spending time on development.  This is because we don't want people to be disappointed if their PR is not merged.
But, if you have already spent the time and developed something, you can open a PR.

The only person who can tell you for sure if something will or will not be merged is jamorham.

**Reply by savek-cc** [reply]
I wrote it as a POC because someone was asking for something like this on facebook. So now it's in my local branch - but I won't put more energy into it as I personally don't require it (much different from e.g. the fix for the duplicate values after sensor change ;)).
